### PR TITLE
Use generic molecule ID in featurization

### PR DESCRIPTION
**Pull request by skearnes**
* Use generic mol_id in featurization output (a prefix can be given with `--mol-prefix`. Note that this change removes the 'names' field from the DataFrame and replaces it with 'mol_id'.
* Make SMILES calculation optional (off by default).

### CODE REVIEW

File: vs_utils/scripts/tests/test_featurize.py [hunk 1]
```diff
@@ -206,6 +207,16 @@ def test_descriptors(self):
         """
         self.check_output(['descriptors'], (2, 196))
 
+    def test_mol_id_prefix(self):
+        """
+        Test that names are prepended with mol_id_prefix.
+        """
+        prefix = 'CID'
+        for i, name in enumerate(self.names):
+          self.names[i] = prefix + name
```
Comment (rbharath) [inline vs_utils/scripts/tests/test_featurize.py]: Maybe rename this field self.mol_id's to match the dataframe.

**Comment by rbharath** [comment]
LGTM after comment has been addressed.

**Comment by skearnes** [comment]
OK, I did a find/replace for names->mol_ids in test_featurize.py and made a corresponding change in featurize.py. Several lines changed in several places, so PTAL.

File: vs_utils/scripts/tests/test_featurize.py [hunk 2]
```diff
@@ -259,12 +270,12 @@ def test_collate_mols1(self):
         """
 
         # write targets
-        targets = {'names': ['ibuprofen'], 'y': [0]}
+        targets = {'mol_id': ['ibuprofen'], 'y': [0]}
```
Comment (rbharath) [inline vs_utils/scripts/tests/test_featurize.py]: s/mol_id/mol_ids/ here and elsewhere.

**Comment by rbharath** [comment]
LGTM. Make sure to swap out mol_id for mol_ids everywhere for consistency.

**Comment by rbharath** [comment]
LGTM.

**Pull request merged**